File: voiture-resevation-frontend/src/app/controller/validators/offre.validator.ts
import {Validator, ValidatorItem} from "@bshg/validation";
import {Offre} from "src/app/controller/entities/offre";

export class OffreValidator extends Validator<Offre> {
  code = new ValidatorItem<string>(
    () => this.item().code,
    (value) => this.item().code = value,
    (value) => {
      this.code.stringValidators
        ?.required()
        ?.valid()
    }
  )

  override validatorItems: ValidatorItem<any>[] = [
    this.code,
  ]

  getValidatorItem(fieldName: string): ValidatorItem<any> {
    // @ts-ignore
    return this[fieldName];
  }

  static init(item: () => Offre): OffreValidator {
    const validator = new OffreValidator()
    validator.item = item
    return validator;
  }

}

File: voiture-resevation-backend/src/main/java/com/example/voitureresevation/entity/validators/OffreValidator.java
package com.example.voitureresevation.entity.validators;

import com.example.voitureresevation.entity.Offre;
import com.example.voitureresevation.zutils.validators.Validator;
import com.example.voitureresevation.zutils.validators.ValidatorItem;

import java.util.List;

public class OffreValidator extends Validator<Offre> {
    public OffreValidator(Offre item) {
        super(item);
    }

    public static void validate(Offre item) {
        new OffreValidator(item).validate();
    }

    public ValidatorItem<String> code = new ValidatorItem<>(
            "code",
            () -> this.getItem().getCode(),
            (String value) -> {
                this.code.getValidators()
                        .required()
                        .valid();
            }
    );

    @Override
    public List<ValidatorItem<?>> validatorItems() {
        return List.of(
                this.code
        );
    }
}
